src/components/GeneralInfo/DataCollectorsCard/DataCollectorsCard.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable camelcase */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import LoadingCard from '../LoadingCard';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+import { TableComposable, Thead, Tr, Th, Tbody, Td, TableVariant, ExpandableRowContent } from '@patternfly/react-table';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { registered } from '../../../Utilities/index';
+
+const getDefaultCollectors = (entity) =>
+    registered?.filter(reporter => reporter.label !== 'insights-client not connected')
+    .map(reporter => ({
+        name: reporter.label,
+        status: entity?.per_reporter_staleness[reporter.value]?.check_in_succeeded,
+        updated: entity?.per_reporter_staleness[reporter.value]?.last_check_in,
+        details: {
+            name: reporter.idName,
+            id: entity?.[reporter.idValue]
+        }
+    }));
+
+const DataCollectorsCard = ({
+    detailLoaded,
+    collectors,
+    entity
+}) => {
+    const [expandedNames, setExpandedNames] = useState([]);
+    const setExpanded = (collector, isExpanding = true) =>
+        setExpandedNames(prevExpanded => {
+            const otherExpandedNames = prevExpanded.filter(r => r !== collector.name);
+            return isExpanding ? [...otherExpandedNames, collector.name] : otherExpandedNames;
+        });
+    const isExpanded = (collector) => expandedNames.includes(collector.name);
+    const data = collectors ?? getDefaultCollectors(entity);
+    return (<LoadingCard
+        title="Data collectors"
+        isLoading={ !detailLoaded }
+    >
+        <TableComposable
+            aria-label="Data collectors"
+            variant={TableVariant.compact}
+            borders={false}
+        >
+            <Thead>
+                <Tr style={{ border: 'none' }}>
+                    <Th />
+                    <Th>{'Name'}</Th>
+                    <Th>{'Status'}</Th>
+                    <Th>{'Last upload'}</Th>
+                </Tr>
+            </Thead>
+            {data.map((collector, rowIndex) => (
```

Comment:
Can we also consume `dataMapper` function which would replace this? Effectively giving consumer an option to completely change the table.
```JS
dataMapper(data, isExpanded, setExpanded)
```